lnwallet/channel.go

```diff
@@ -4331,6 +4331,14 @@ func (lc *LightningChannel) ReceiveHTLC(htlc *lnwire.UpdateAddHTLC) (uint64, err
 			"ID %d", htlc.ID, lc.remoteUpdateLog.htlcCounter)
 	}
 
+	// Clamp down on the number of HTLC's we can receive. It should never
+	// exceed our configurable MaxAcceptedHTLCs.
+	maxHTLCs := lc.channelState.LocalChanCfg.MaxAcceptedHtlcs
+	remoteACKedIndex := lc.localCommitChain.tail().theirHtlcIndex
```

Comment:
This will be the index of the last acked htlc,  not the first one active, which results in checking the number of unACKed HTLCs, which I don't think is what we want?

What we want to check is the number of active HTLCs, which will include both ACKed and unACKed. I think we must check the update logs or the commitments to see which HTLCs are actually active.

Reply:
Ahhh you're right